libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -467,6 +467,57 @@ void WaveTrack::SetRate(double newRate)
    SetClipRates(newRate);
 }
 
+bool WaveTrack::GetFloatsCenteredAround(
+   double t, size_t iChannel, float* buffer, size_t numSideSamples) const
+{
+   const auto clip = GetClipAtTime(t);
+   if (clip == nullptr)
+      return false;
+   // Assuming an odd bufferLen. If it's not, then the result may not be quite
+   // as expected, but that's all right.
+   const auto maybeNegativeStart =
+      clip->TimeToSamples(t - clip->GetPlayStartTime()) -
+      sampleCount { numSideSamples };
+   const auto start = std::max(sampleCount { 0 }, maybeNegativeStart);
+   const auto numLeadingZeros = (start - maybeNegativeStart).as_size_t();
+   std::fill(buffer, buffer + numLeadingZeros, 0.f);
+   auto offsetBuffer = reinterpret_cast<samplePtr>(buffer + numLeadingZeros);
+   const auto len = numSideSamples * 2 + 1u - numLeadingZeros;
+   if (start + len > clip->GetVisibleSampleCount())
+      return false;
+   clip->GetSamples(iChannel, offsetBuffer, floatSample, start, len);
+   return true;
```

Comment:
Some edge-case behavior of the paintbrush tool will change.

First, maybe two clips are touching, you click just right of the boundary, and where before the left hand clip might be changed too, now it isn't.

Secondly, this routine has an asymmetry.  Fetching some initial out-of-bounds zeroes lets it succeed, but you fail instead of fetching right.

To be really conservative, getting and setting of samples from neighboring clips, when there are only unit sample rations (because the user doesn't use tempo or time stretch at all), should work.

But this is a very edgy edge-case in a little known and used feature.